tests/go-validation/v1apitests/provisioning_test.go

```diff
@@ -0,0 +1,444 @@
+	adminToken string
+	userToken  string
+	cni        string
+	k8sVersion string
+
+	// Digital Ocean variables
+	digitalOceanCloudCredential       *cloudcredentials.CloudCredential
+	digitalOceanCloudCredentialConfig *corevV1.Secret
+
+	// Node Driver variables
+	rkeMachinePools []apisV1.RKEMachinePool
+	machineConfig   *unstructured.Unstructured
+
+	// AWS environment variables
+	awsInstanceType    string
+	awsRegion          string
+	awsRegionAZ        string
+	awsAMI             string
+	awsSecurityGroup   string
+	awsSSHKeyName      string
+	awsCICDInstanceTag string
+	awsIAMProfile      string
+	awsUser            string
+	awsVolumeSize      int
+
+	ec2Client        *aws.EC2Client
+	nodes            []*aws.EC2Node
+	nodeConfig       *cluster.NodeConfig
+	nodeRoles        []string
+	baseEC2Name      string
+	baseClusterName  string
+	clusterNamespace string
+	randString       string
+
+	// rest api variables
+	bearerName       string
+	bearerToken      string
+	managementClient *v3.Client
+
+	clusterNames []string
+	clusterObjs  []*cluster.Cluster
+	v1Cluster    *apisV1.Cluster
+
+	rancherCleanup bool
+}
+
+// The SetupSuite method will be run by testify once, at the very
+// start of the testing suite, before any tests are run.
+func (s *ProvisioningTestSuite) SetupSuite() {
+	s.host = clients.Host
+	s.password = user.Password
+	s.adminToken = clients.AdminToken
+	s.userToken = clients.UserToken
+	s.cni = cluster.CNI
+	s.k8sVersion = cluster.KubernetesVersion
+	s.clusterNamespace = "fleet-default"
+	s.rancherCleanup = environmentvariables.RancherCleanup()
+
+	testifyArg := os.Args[5:len(os.Args)]
+
+	var testCase string
+	if testifyArg[0] == "-testify.m" {
+		testCase = testifyArg[1]
+	} else {
+		testCase = ""
+	}
+
+	if strings.Contains(testCase, "DigitalOcean") || testCase == "" {
+		for userName, bearerToken := range clients.BearerTokensList() {
+
+			s.T().Logf("Create Cloud Credential for user %s", userName)
+			client, err := clients.NewCoreV1Client(bearerToken)
+			require.NoError(s.T(), err)
+
+			cloudCredentialName := digitalOceanCloudCredentialName + namegenerator.RandStringLowerBytes(defaultRandStringLength)
+			doCloudCred := cloudcredentials.NewCloudCredentialSecret(cloudCredentialName, "", "digitalocean", namespace)
+
+			cloudCredential := cloudcredentials.NewCloudCredential(client)
+			_, err = cloudCredential.CreateCloudCredential(doCloudCred)
+			require.NoError(s.T(), err)
+
+			s.digitalOceanCloudCredential = cloudCredential
+			s.digitalOceanCloudCredentialConfig = doCloudCred
+		}
+	}
+	if strings.Contains(testCase, "CustomCluster") || testCase == "" {
+		s.awsInstanceType = aws.AWSInstanceType
+		s.awsRegion = aws.AWSRegion
+		s.awsRegionAZ = aws.AWSRegionAZ
+		s.awsAMI = aws.AWSAMI
+		s.awsSecurityGroup = aws.AWSSecurityGroup
+		s.awsSSHKeyName = aws.AWSSSHKeyName
+		s.awsCICDInstanceTag = aws.AWSCICDInstanceTag
+		s.awsIAMProfile = aws.AWSIAMProfile
+		s.awsUser = aws.AWSUser
+		s.awsVolumeSize = aws.AWSVolumeSize
```

Comment:
I'm doing this in CreateEC2Instances() line 49. I think this should be done every time ec2 instances are created.

Reply:
I must've commented on this before refreshing the page or something since I think we weren't using `s.ec2Client` at all before. Should `CreateEC2Instances()` be part of the Suite setup instead of being called by individual Tests?

Reply:
I don't think so, since we want to call this whenever we're provisioning clusters, I think it's fine where it's at.